content/developer/reference/frontend/owl_components.rst

```diff
@@ -150,13 +150,144 @@ checkboxes or datepickers. This page explains how to use these generic component
    * - :ref:`CheckBox <frontend/owl/checkbox>`
      - a simple checkbox component with a label next to it
    * - :ref:`Dropdown <frontend/owl/dropdown>`
      - full-featured dropdown
    * - :ref:`Pager <frontend/pager>`
      - a small component to handle pagination
 
+.. _frontend/owl/actionswiper:
+
+ActionSwiper
+------------
+
+.. note:: This component is a mobile feature, only supported in the Enterprise version of Odoo.
+
+Location
+~~~~~~~~
+
+`@web_enterprise/core/action_swiper/action_swiper`
+
+Description
+~~~~~~~~~~~
+
+This is a component that can perform actions when an element is swiped
+horizontally. The swiper is wrapping a target element to add actions to it. 
+The action is executed once the user has released the swiper passed
+half of its width.
+
+.. code-block:: xml
+
+  <ActionSwiper onLeftSwipe="Object" onRightSwipe="Object">
+    <SomeElement/>
+  </ActionSwiper>
+
+The simpliest way to use the component is to use it around your target element directly
+in an xml template as shown above. But sometimes, you my want to extend an existing element
+and would not want to duplicate the template. It is possible to do just that.
+
+If you want to extend the behavior of an existing element, you must place the element
+inside, by wrapping it directly. Also, you can conditionnally add props to manage when the
+element might be swipable or not. 
+
+You can use the component to interact easily with records, messages, items in lists and much more.
+
+.. note:: Actions are permuted when using right-to-left (RTL) languages.
+
+Props
+~~~~~
+
+.. list-table::
+    :widths: 20 20 60
+    :header-rows: 1
+
+    * - Name 
+      - Type
+      - Description
+    * - `onLeftSwipe`
+      - `Object`
+      - if present, the actionswiper can be swiped to the left
+    * - `onRightSwipe`
+      - `Object`
+      - if present, the actionswiper can be swiped to the right
+
+You can use both ``onLeftSwipe`` and ``onRightSwipe`` props at the same time.
+
+Those ``Object``'s must contain:
+    
+    - ``action``, which is the callable ``Function`` serving as a callback.
+      Once the swipe has been completed in the given direction, that action
+      is performed.
+    - ``icon`` is the icon class to use, usually to represent the action.
+      It must be a ``string``.
+    - ``bgColor`` is the background color, given to decorate the action.
+      can be one of the following `bootstrap contextual color <https://getbootstrap.com/docs/3.3/components/#available-variations>`_ (``danger``,
+      ``info``, ``secondary``, ``success`` or ``warning``).
+
+    Those values must be given to define the behavior and the visual aspect
+    of the swiper. 
+
+Example: Create a basic swipable element
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This first example uses some basic features of the ActionSwiper component. 
+Here, the swipe is enabled in both directions.
+
+.. code-block:: xml
+    
+  <ActionSwiper 
```

Comment:
move this in the description section

Reply:
Sure it's better? I'd prefer consistency, the Dropdown puts the examples at the end of the section: 
![image](https://user-images.githubusercontent.com/35101914/145960433-e7b1bce1-b3d8-4552-85e0-abcd9276f316.png)

Reply:
I've put the examples in the Description, but I'm not sure it's better like this

Reply:
that's for advanced usecase. I think the best is to have a simple (realistic) example in the description, and if needed, a few more advanced at the end